tests/eventdispatcher/PatientFilterEventHookTest/acl/acl_upgrade.php

```diff
@@ -0,0 +1,46 @@
+<?php
+// Ensure this script is not called separately
+if ((empty($_SESSION['lang_module_unique_id'])) ||
+    (empty($unique_id)) ||
+    ($unique_id != $_SESSION['lang_module_unique_id'])) {
+    die('Authentication Error');
+}
+unset($_SESSION['lang_module_unique_id']);
+
+use OpenEMR\Common\Acl\AclExtended;
+
+function upgradeAclFromVersion($version)
+{
+    global $ACL_UPGRADE;
+    $toVersion = '';
+    foreach ($ACL_UPGRADE as $toVersion => $function) {
+        if (version_compare($version, $toVersion) < 0) {
+            $function();
+        }
+    }
+    return $toVersion;
+}
```

Comment:
Is there a way to migrate this function out of this script (just don't see a reason for the "wiring" of this feature to be actually included within the acl_upgrade script, which means it gets replicated in all the other modules and will be impossible to improve/fix it in the future). Seems like all this uprade script needs to do is populate the $ACL_UPGRADE variable.

Reply:
I tried to make the function to the class InstallerController